http/file_server_test.ts

```diff
@@ -1,51 +1,42 @@
-import { readFile } from "deno";
+import { readFile, run } from "deno";
 
-import { test, assert, assertEqual } from "../testing/mod.ts";
+import { test, assert, assertEqual, setUp, tearDown } from "../testing/mod.ts";
 
-// Promise to completeResolve when all tests completes
-let completeResolve;
-export const completePromise = new Promise(res => (completeResolve = res));
-let completedTestCount = 0;
-
-function maybeCompleteTests() {
-  completedTestCount++;
-  // Change this when adding more tests
-  if (completedTestCount === 3) {
-    completeResolve();
-  }
-}
-
-export function runTests(serverReadyPromise: Promise<any>) {
-  test(async function serveFile() {
-    await serverReadyPromise;
-    const res = await fetch("http://localhost:4500/azure-pipelines.yml");
-    assert(res.headers.has("access-control-allow-origin"));
-    assert(res.headers.has("access-control-allow-headers"));
-    assertEqual(res.headers.get("content-type"), "text/yaml; charset=utf-8");
-    const downloadedFile = await res.text();
-    const localFile = new TextDecoder().decode(
-      await readFile("./azure-pipelines.yml")
-    );
-    assertEqual(downloadedFile, localFile);
-    maybeCompleteTests();
+var fileServer;
+setUp(async function startFileServer() {
+  fileServer = run({
+    args: ["deno", "--allow-net", "http/file_server.ts", ".", "--cors"]
   });
+  // Wait for fileServer to spool up.
+  await new Promise(res => setTimeout(res, 1000));
```

Comment:
We really need a better way to do this...

Reply:
This can be done pretty easily with these utils:
```ts
setUp(async function startFileServer() {
  fileServer = run({
    args: ["deno", "--allow-net", "http/file_server.ts", ".", "--cors"]
  });
  // Wait for fileServer to spool up.
  // await new Promise(res => setTimeout(res, 1000));
  await waitUntil(async function() {
    try { await fetch("http://localhost:4500/"); return true }
    catch(e) { return false } ;
  }, 1000)
});

// given
async function waitUntil(isReady: ()=>boolean|Promise<boolean>, timeout: number): Promise<void> {
  let t = 0;
  while (!await isReady()) {
    await delay(1000);
    if (t++ > timeout) {
      throw Error(`Timeout waiting for ${isReady}`)
    }
  }
}

async function delay(ms: number): Promise<void> {
  await new Promise(res => setTimeout(res, ms));
}
```
do `waitUntil` and `delay` belong in an std as standalone?

I can update this PR...

Reply:
I think a better way to wait for file_server to be ready is to read from its stdout and wait until you see "HTTP server listening on http://0.0.0.0:4500/"

Reply:
Fixed.